File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/context/julia/runtime/contract/InputsAndResultsTestDecimal.jl
println("getDecimal")
if isdefined(Main, :getDecimal) && !isnothing(getDecimal)
	error("getDecimal already defined!")
end
getDecimal = putDecimal
println(typeof(getDecimal))
println(getDecimal)
if typeof(getDecimal) != Float64
	error("getDecimal not Float64!")
end

println("getDecimalVector")
if isdefined(Main, :getDecimalVector) && !isnothing(getDecimalVector)
	error("getDecimalVector already defined!")
end
getDecimalVector = putDecimalVector
println(eltype(getDecimalVector))
println(getDecimalVector)
if eltype(getDecimalVector) != Float64
	error("getDecimalVector not Float64!")
end

println("getDecimalVectorAsList")
if isdefined(Main, :getDecimalVectorAsList) && !isnothing(getDecimalVectorAsList)
	error("getDecimalVectorAsList already defined!")
end
getDecimalVectorAsList = putDecimalVectorAsList
println(eltype(getDecimalVectorAsList))
println(getDecimalVectorAsList)
if eltype(getDecimalVectorAsList) != Float64
	error("getDecimalVectorAsList not Float64!")
end

println("getDecimalMatrix")
if isdefined(Main, :getDecimalMatrix) && !isnothing(getDecimalMatrix)
	error("getDecimalMatrix already defined!")
end
getDecimalMatrix = putDecimalMatrix
println(eltype(getDecimalMatrix))
println(getDecimalMatrix)
if eltype(getDecimalMatrix) != Float64
	error("getDecimalMatrix not Float64!")
end

println("getDecimalMatrixAsList")
if isdefined(Main, :getDecimalMatrixAsList) && !isnothing(getDecimalMatrixAsList)
	error("getDecimalMatrixAsList already defined!")
end
getDecimalMatrixAsList = putDecimalMatrixAsList
println(eltype(getDecimalMatrixAsList))
println(getDecimalMatrixAsList)
if eltype(getDecimalMatrixAsList) != Float64
	error("getDecimalMatrixAsList not Float64!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestDecimal.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getDecimal")
if(getDecimal ~= nil) then
	error("getDecimal already defined!")
end
getDecimal = putDecimal
getDecimalType = type(getDecimal)
print(getDecimalType)
print(getDecimal)
if(getDecimalType ~= "number") then
	error("getDecimal not Double!")
end

print("getDecimalVector")
if(getDecimalVector ~= nil) then
	error("getDecimalVector already defined!")
end
getDecimalVector = putDecimalVector
getDecimalVectorType = type(getDecimalVector[1])
print(getDecimalVectorType)
print(arraysToString(getDecimalVector))
if(getDecimalVectorType ~= "number") then
	error("getDecimalVector not Double!")
end

print("getDecimalVectorAsList")
if(getDecimalVectorAsList ~= nil) then
	error("getDecimalVectorAsList already defined!")
end
getDecimalVectorAsList = putDecimalVectorAsList
getDecimalVectorAsListType = type(getDecimalVectorAsList[1])
print(getDecimalVectorAsListType)
print(arraysToString(getDecimalVectorAsList))
if(getDecimalVectorAsListType ~= "number") then
	error("getDecimalVectorAsList not Double!")
end

print("getDecimalMatrix")
if(getDecimalMatrix ~= nil) then
	error("getDecimalMatrix already defined!")
end
getDecimalMatrix = putDecimalMatrix
getDecimalMatrixType = type(getDecimalMatrix[1][1])
print(getDecimalMatrixType)
print(arraysToString(getDecimalMatrix))
if(getDecimalMatrixType ~= "number") then
	error("getDecimalMatrix not Double!")
end

print("getDecimalMatrixAsList")
if(getDecimalMatrixAsList ~= nil) then
	error("getDecimalMatrixAsList already defined!")
end
getDecimalMatrixAsList = putDecimalMatrixAsList
getDecimalMatrixAsListType = type(getDecimalMatrixAsList[1][1])
print(getDecimalMatrixAsListType)
print(arraysToString(getDecimalMatrixAsList))
if(getDecimalMatrixAsListType ~= "number") then
	error("getDecimalMatrixAsList not Double!")
end
